Brokerages/Exante/ExanteBrokerage.DataQueueHandler.cs

```diff
@@ -0,0 +1,256 @@
+                        var feedQuoteStream = _client.StreamClient.GetFeedQuoteStreamAsync(
+                            new[] {ticker},
+                            tickShort =>
+                            {
+                                var tick = CreateTick(tickShort);
+                                if (tick != null)
+                                {
+                                    Aggregator.Update(tick);
+                                }
+                            },
+                            level: ExanteQuoteLevel.BestPrice).SynchronouslyAwaitTaskResult();
+                        if (!feedQuoteStream.Success)
+                        {
+                            Log.Error(
+                                $"Exante.StreamClient.GetFeedQuoteStreamAsync({ticker}): " +
+                                $"Error: {feedQuoteStream.Error}"
+                            );
+                        }
+
+                        var feedTradesStream = _client.StreamClient.GetFeedTradesStreamAsync(
+                            new[] {ticker},
+                            feedTrade =>
+                            {
+                                var tick = CreateTick(feedTrade);
+                                if (tick != null)
+                                {
+                                    Aggregator.Update(tick);
+                                }
+                            }).SynchronouslyAwaitTaskResult();
+                        if (!feedTradesStream.Success)
+                        {
+                            Log.Error(
+                                $"Exante.StreamClient.GetFeedTradesStreamAsync({ticker}): " +
+                                $"Error: {feedTradesStream.Error}"
+                            );
+                        }
+
+                        _subscribedTickersStreamSubscriptions[ticker] = (feedQuoteStream.Data, feedTradesStream.Data);
+                    }
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Create a tick from the Exante feed stream data
+        /// </summary>
+        /// <param name="exanteFeedTrade">Exante feed stream data object</param>
+        /// <returns>LEAN Tick object</returns>
+        private Tick CreateTick(ExanteFeedTrade exanteFeedTrade)
+        {
+            var symbolId = exanteFeedTrade.SymbolId;
+            if (!_subscribedTickers.TryGetValue(symbolId, out var symbol))
+            {
+                // Not subscribed to this symbol.
+                return null;
+            }
+
+            if (exanteFeedTrade.Size == decimal.Zero)
+            {
+                return null;
+            }
+
+            var utc = exanteFeedTrade.Date;
+            // Convert the timestamp to exchange timezone and pass into algorithm
+            var time = utc.ConvertTo(DateTimeZone.Utc, TimeZones.NewYork);
+
+            var instrument = _client.GetSymbol(symbolId);
```

Comment:
Should probably cache this guy out `var instrument = _client.GetSymbol(symbolId);`? This might we called many times per second, doing this call will add a significant overhead

Reply:
ping